index.bs

```diff
@@ -717,6 +722,58 @@ is done in a consistent and privacy-friendly way:
      stable id during the session for the given origin, but also be able to deterministically produce
      the same id in subsequent sessions.
 
+<h3 id="wrapper-apis">Native APIs don't typically translate well to the web</h3>
+
+Many modern operating systems come with convenience APIs, which abstract away
+normally complex technology in the form of higher level user-friendly APIs.
+
+Exposing these APIs to the web platform is getting more and more common, but you
+need to be careful when exposing these to the web platform.
+
+ : Generalize interface when underlying API is overly specific
+ :: Don't blindly map native convenience APIs 1:1 to the web platform.
+    Native APIs do not translate well to the web platform, and doing so may result
+    in a API shape which is difficult to implement across a variety of underlying
+    platforms and native libraries.
+
+    Especially be careful about exposing the exact lifecycle and data
+    structures of the underlying native APIs. Think about making an API that
+    fits well in with existing web platform APIs and which can be expressed in
+    commonly available low-level APIs.
+
+ : Prefer asynchronous function by returning promises
+ :: Even if the underlying API is synchronous, this does not necessarily mean it
+    translates well when ported to the web platform. Exposing a synchronous native
+    API to the web platform is generally discouraged, and should be
+    provided as a promise based API whenever possible.
+
+    Doing this also allows putting the API behind a permission if it is
+    necessary.
+
+ : Prevent underlying OS fragmentation
```

Comment:
I don't think this header is right... maybe: "Don't overconstrain or overfit the API to the underlying (wrapped) OS